### DISCUSSION TITLE
[React JSX] Why optional properties have both `?:`  and `| undefined` ( see example)

### DISCUSSION THREAD

**Discussion by MananTank** [DefinitelyTyped/DefinitelyTyped#61328] [_Misc]
<img width="640" alt="image" src="https://user-images.githubusercontent.com/22043396/179661658-28c791b8-114a-473d-a015-e607f5f2499e.png">

Is the `| undefined` necessary here?
Can they be removed?

**Comment by BendingBender** [comment]
This is needed because of the [`exactOptionalPropertyTypes`](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes) compiler flag. When this flag was introduced, the whole DT repo was converted to use both, the optional qualifier and `| undefined` so that types don't break when this compiler option is enabled.

Strictly seen, you need to check whether properties actually don't exist on an object or are set expicitly to `undefined`, then you can adapt the definitions accordingly, depending on what actually happens in the JS code. But in my opinion, it's just not worth it, it's a very subtle distinction and for the most time it just doesn't make any difference.

I usually stick to using both, the optional qualifier and `| undefined` to not trip up users that have this compiler flag enabled.
